### PR TITLE
added fullAddress

### PR DESCRIPTION
**Pull request by sgrg1**
added fullAddress

### CODE REVIEW

File: src/components/trashlookup.jsx [hunk 1]
```diff
@@ -47,6 +47,7 @@ class TrashLookUp extends Component {
 
     if (searchQuery.length > 0) {
       filtered = PostData.filter(m => m.address1.toLowerCase().indexOf(searchQuery.toLowerCase()) > -1);
+      console.log(filtered);
```
Comment (martypowell) [inline src/components/trashlookup.jsx]: please remove console.log, this should only be used for debugging.

File: src/components/trashlookup.jsx [hunk 2]
```diff
@@ -100,13 +101,11 @@ class TrashLookUp extends Component {
...
           .format(dateFormat);
   }
-
   render() {
     let isAutoTextHidden = this.state.isAutoTextHidden;
     const selectedAddress = _.trim(this.state.selectedAddress);
     const { data } = this.addressData(isAutoTextHidden);
-
-
+    const fullAddress = data.length > 0 ?`${data[0].address1}${`, `} ${data[0].city}${`, `} ${data[0].state}${` `} ${data[0].postalCode}`:'none';
```
Comment (martypowell) [inline src/components/trashlookup.jsx]: Can you please create a folder called `Services` and inside that folder move this logic to a file called `AddressService.js`. Inside the file you will want to do something like

```js
const parseAddress = address => {
  return data.length > 0 ?`${data[0].address1}${`, `} ${data[0].city}${`, `} ${data[0].state}${` `} ${data[0].postalCode}`:'none'
}

export {
  parseAddress
}
```

in this file you would then use it this way:

at the top

```js
import { parseAddress } from './Services/AddressService.js'
```

then you can call it like this
```js
const fullAddress = parseAddress(data);
```

In the services folder you should also add a file `AddressService.test.js` and inside that file you should add some tests for your parseAddress function. Once you get to this point, let me know if you need help.

**Review by martypowell — CHANGES_REQUESTED** [review]

File: src/components/Services/AddressService.test.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import { parseAddress } from './AddressService'
+
+
+const data = [{ address1: "1745 T Street Southeast1", city: "Louisville", state: "KY", postalCode: "40219" }]
+
+it('passing address data', () => {
```
Comment (martypowell) [inline src/components/Services/AddressService.test.js]: think you could use a describe block here

```js
describe('parseAddress', () => {
  // tests go here
});

Comment (martypowell) [reply src/components/Services/AddressService.test.js]: i don't have this documented but would like to try and follow this format

```js
it('should return a friendly message when the address is empty', () => {
  // arrange
  const emptyAddress = {};
 
  // act
  const actual = getFullAddress(emptyAddress);

  // assert
  expect(actual).toEqual('Address was null or empty');
});
```

**Review by martypowell — CHANGES_REQUESTED** [review]

File: src/services/AddressService.test.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    it('should return a friendly message when the address is empty', () => {
+        //arrange
+        const emptyAddress = {};
+
+        //act
+        const actual = getFullAddress(emptyAddress);
+
+        //assert
+        expect(actual).toEqual(
+            '1745 T Street Southeast1, Louisville, KY  40219'
```
Comment (martypowell) [inline src/services/AddressService.test.js]: your test says that you will return a friendly message when the address is empty, but it looks like you are returning an actual address?

File: src/services/AddressService.test.js [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+        );
+    });
+
+    it('should return a friendly message when the address is matched', () => {
```
Comment (martypowell) [inline src/services/AddressService.test.js]: i think in this test case you are saying

"should return the full address when all address properties are provided"

File: src/services/AddressService.test.js [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+        //arrange
+        const emptyAddress = [
```
Comment (martypowell) [inline src/services/AddressService.test.js]: the variable name to me suggests that the address would be empty, but you have a valid address?

File: src/services/AddressService.jsx [hunk 7]
```diff
@@ -0,0 +1,10 @@
+const getFullAddress = address => {
```
Comment (martypowell) [inline src/services/AddressService.jsx:3]: after looking at your tests, it looks like address you are passing in array, but you should only need to pass in the object

example

currently 
```js
[{
  address1: "123 Abc Ave"
}]
```

proposed

```js
{
  address1: "123 Abc Ave"
}
```

Comment (martypowell) [reply src/services/AddressService.jsx:3]: this will impact your tests, so they will need to be updated

File: src/services/AddressService.test.js [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+        //act
+        const actual = getFullAddress(emptyAddress);
+
+        //assert
+        expect(actual).toEqual(
+            '1745 T Street Southeast1, Apt 15, Louisville, KY 40219'
+        );
+    });
+
+    it('should return a friendly message when address does not match', () => {
```
Comment (martypowell) [inline src/services/AddressService.test.js]: i think in this test case you are saying

"should return the full address when all address properties are provided except address2"

File: src/services/AddressService.test.js [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+        //arrange
+        const emptyAddress = [
```
Comment (martypowell) [inline src/services/AddressService.test.js]: the variable name to me suggests that the address would be empty, but you have a valid address?

**Review by martypowell — DISMISSED** [review]

File: src/components/trashlookup.jsx (hunk 1 above)
Comment (sgrg1) [reply src/components/trashlookup.jsx]: console.log is removed

File: src/components/trashlookup.jsx (hunk 2 above)
Comment (sgrg1) [reply src/components/trashlookup.jsx]: services folder is created and called getFullAddress(Address) control

File: src/services/AddressService.jsx (hunk 7 above)
Comment (sgrg1) [reply src/services/AddressService.jsx:3]: change code to pass in the object instead of array

File: src/services/AddressService.test.js (hunk 8 above)
Comment (sgrg1) [reply src/services/AddressService.test.js]: fixed issue with testing

File: src/components/trashlookup.jsx [hunk 10]
```diff
@@ -7,123 +6,150 @@ import { getTrashService } from '../services/trashService';
 import * as moment from 'moment';
 import TrashSchedule from './common/trashSchedule';
 import RenderList from './common/renderList';
+import { getFullAddress } from '../services/AddressService';
 class TrashLookUp extends Component {
```
Comment (martypowell) [inline src/components/trashlookup.jsx:10]: Refactor work captured here #16

File: src/services/AddressService.jsx [hunk 11]
```diff
@@ -0,0 +1,15 @@
+import _ from "lodash";
+
+const getFullAddress = address => {
+  return !_.isEmpty(address)
```
Comment (martypowell) [inline src/services/AddressService.jsx:4]: this should be refactored, using object destructuring. not blocking, will submit another pr for this.

File: src/services/AddressService.test.js [hunk 12]
```diff
@@ -0,0 +1,50 @@
+import { getFullAddress } from "./AddressService";
```
Comment (martypowell) [inline src/services/AddressService.test.js:1]: Exported functions should be UpperCase, not blocking, will updated in another pr.

**Review by martypowell — APPROVED** [review]
Some works needs to be done with this, but we will do it in the future.
